### ISSUE TITLE
Front-end rewrite in React.js

### ISSUE THREAD

**Issue by tschoffelen** [rchain-community/rchain.cloud#30]
Now that we've started rapidly adding features to the front-end of rchain.cloud (mostly thanks to @aostrun, @AhsanFazal and others), it makes sense to start using a stable framework to build out those features. 

That will allow us to use routing, componentize UI elements, and decrease the chance of merge conflicts by not having to do all the Javascript work in the same file. 😄 

## New branch

I created a new branch called [`react`](https://github.com/th3build/rchain.cloud/tree/react) that we can use to build those new features, to make sure that we keep a stable version in `master` until we get at least the basics up and running in React.

## Way of working

To make sure we don't do double work, please always open a Github issue before starting to work on something. This also allows the team to jump in and discuss early if we aren't completely sure if the thing you're planning to develop is the right thing to spend time on right now.

Whenever you actually start writing code, __make sure the issue is assigned to yourself__! You can do so either by assigning yourself if you're a collaborator of this repo, or by asking myself or @AhsanFazal to do so for you. That way everyone can directly see what the others in the team are working on. Thanks!

## Code style

Some general notes from my side:

* Let's add a .eslintrc file that uses `standard` & `standard-react` as code style formatters.
* Also, when building the new React app, let's make sure to keep components under a 100 lines long. If a file is longer than 100 lines, you probably need to split it up.

## Main priorities

- [x] #20 Set up Webpack and modify the Express app to serve the new React/Webpack app.
- [x] #21 Set up `react-router`. For now there will only be one route (`/`), but I see this quickly expanding as new features are being added.
- [x] #22 Set up `eslint` for this project.
- [ ] #23 Manage websocket client connecting in a React HOC component.
- [x] #24 Rebuild main UI (navigation bar, etc.) as React components.
- [x] #25 Rebuild code editor as React component.
- [x] #26 Rebuild output formatter as React component.
- [x] #27 Rebuild examples menu as React component.
- [ ] #28 Write some basic tests that we can use with a CI pipeline.
- [ ] #29 Set up CI pipeline for automatically deploying updates.

Feel free to assign yourself to any of these, and let's get this going! 

We need to have some coordination, because some of the items above need to be done in a certain order. Let's try to keep our discussion visible to everyone by either adding comments to this issue, or to one of the issues listed above. 🚀
